File: Backend/models/deck.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const DeckSchema = new Schema({
    _id: mongoose.Schema.Types.ObjectId,
    userId: {
        type: String,
        required: [true, 'User ID is required'],
    },
    name: {
        type: String,
        required: [true, "Deck name is required"],
    },
    cardArt: {
        type: String
    },
    mainDeckCount: {
        type: Number,
        default: 0,
        min: 0,
        max: 60
    },
    extraDeckCount: {
        type: Number,
        default: 0,
        min: 0,
        max: 15
    },
    sideDeckCount: {
        type: Number,
        default: 0,
        min: 0,
        max: 15
    },
});

const Deck = mongoose.model('deck', DeckSchema);
module.exports = Deck;

File: Frontend/flutter_app/lib/models/deck.dart
class Deck {
  String? sId;
  String? userId;
  String? name;
  int? mainDeckCount;
  int? extraDeckCount;
  int? sideDeckCount;

  Deck(
      {this.sId,
      this.userId,
      this.name,
      this.mainDeckCount,
      this.extraDeckCount,
      this.sideDeckCount});

  Deck.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    userId = json['userId'];
    name = json['name'];
    mainDeckCount = json['mainDeckCount'];
    extraDeckCount = json['extraDeckCount'];
    sideDeckCount = json['sideDeckCount'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = sId;
    data['userId'] = userId;
    data['name'] = name;
    data['mainDeckCount'] = mainDeckCount;
    data['extraDeckCount'] = extraDeckCount;
    data['sideDeckCount'] = sideDeckCount;
    return data;
  }
}
